src/compiler/checker.ts

```diff
@@ -13929,6 +13929,17 @@ namespace ts {
                             Diagnostics.The_expected_type_comes_from_the_return_type_of_this_signature,
                         ));
                     }
+                    if ((getFunctionFlags(node) & FunctionFlags.Async) === 0
+                        // exclude cases where source itself is promisy - this way we don't make a suggestion when relating
+                        // an IPromise and a Promise that are slightly different
+                        && !getTypeOfPropertyOfType(sourceReturn, "then" as __String)

```

Comment:
I'm surprised we don't have a helper for this.

Reply:
This meaning...? Seeing if a type is promisish (highly non-technical term)? Yeah, I went looking inside `getAwaitedType` to see if we had a helper and saw that all we did was essentially look for a `then` property (to match spec behavior, ofc), so that's what I did here. A helper would essentially just curry one argument and cast the result, so it doesn't seem terribly needed.

Reply:
I meant checking for a `then` property (don't we have a `Thenable` concept?).

Reply:
Nope. `Thenable` is just a Promises/A* spec thing that roughly maps to "object with a then property". We have no interface or anything special for that.